### PR TITLE
Dictionary Attributes with new lines.

### PR DESCRIPTION
**Pull request by sezgin1947**

### CODE REVIEW

**Comment by mgehre-amd** [comment]
Could you please add a test case to show the capabilities (e.g. in mlir/test/IR/mlir-newline-after-attr.mlir)?

File: mlir/lib/IR/AsmPrinter.cpp [hunk 1]
```diff
@@ -2605,7 +2612,8 @@ void AsmPrinter::Impl::printOptionalAttrDict(ArrayRef<NamedAttribute> attrs,
 
     // Otherwise, print them all out in braces.
     interleave(
-        filteredAttrs, [&](NamedAttribute attr) { printNamedAttribute(attr); },
+        filteredAttrs,
+        [&](NamedAttribute attr) { printNamedAttribute(attr, separator); },
```
Comment (mgehre-amd) [inline mlir/lib/IR/AsmPrinter.cpp:2616]: For the inner separator, don't you need to add two extra spaces to make them visually indented?

Comment (sezgin1947) [reply mlir/lib/IR/AsmPrinter.cpp:2616]: I initially implemented this. However, a modification on seperator here applies for every dictionary attribute (that is why seperation information passed from `AsmPrinter::Impl::printOptionalAttrDict` explicitly, while other calls to `AsmPrinter::Impl::printAttributeImpl` are not changed and using the default value for `seperator` argument)
 Consequently, we get the following IR if we add extra spaces here:
image

Comment (mgehre-amd) [reply mlir/lib/IR/AsmPrinter.cpp:2616]: Right, you only want to add two extra spaces here if the 
```
if (printerFlags.getNewlineAfterAttrLimit() &&
        attrs.size() > *printerFlags.getNewlineAfterAttrLimit()) {
```
branch was taken above

Comment (mgehre-amd) [reply mlir/lib/IR/AsmPrinter.cpp:2623]: Could you please add a test case to show the capabilities (e.g. in mlir/test/IR/mlir-newline-after-attr.mlir)?

**Review by mgehre-amd — APPROVED** [review]
Great!

**Pull request merged**